### PR TITLE
Move some common util functions to an individual file

### PR DESCRIPTION
**Pull request by m1093782566**
There are some duplicated installation scripts among minikube and vagrant. This PR refactor some common scripts into functions and move them to an individual file - common.sh

### CODE REVIEW

**Comment by m1093782566** [comment]
Kindly ping @costinm @baodongli @JimmyCYJ @gargnupur @hklai for review :)

**Comment by m1093782566** [comment]
@costinm @yutongz Would you please help review this PR when you have chance?

File: tests/e2e/local/common.sh [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+# limitations under the License.
+
+# Common utilities for Istio local installation.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Install Curl on Ubuntu platform
+function install_curl_ubuntu() {
```
Comment (hklai) [inline tests/e2e/local/common.sh]: I love moving all these to functions so they can be reused.

However, could you separate linux/macos utils into different files for better readability?

File: tests/e2e/local/common.sh [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  fi
+}
+
+# Check if homebrew is installed
+function check_homebrew() {
+  if ! brew --help > /dev/null; then
+    echo "Homebrew is not installed. Please go to https://docs.brew.sh/Installation to install Homebrew."
+    exit 1
+  fi
+}
```
Comment (baodongli) [inline tests/e2e/local/common.sh]: A single function like check_install_cmd() with two arguments 'cmd' and 'install_url", should cover brew, apt-get, dpkg.

File: tests/e2e/local/common.sh (hunk 1 above)
Comment (m1093782566) [reply tests/e2e/local/common.sh]: @hklai 

> could you separate linux/macos utils into different files for better readability?

Make sense, it's fixed now. Please take a look when you have chance :) THANKS!

**Comment by m1093782566** [comment]
Awesome! Thanks all :)

**Pull request merged**